MediaBrowser.Api/BaseApiService.cs

```diff
@@ -16,30 +18,43 @@ namespace MediaBrowser.Api
     /// <summary>
     /// Class BaseApiService
     /// </summary>
-    public class BaseApiService : IService, IRequiresRequest
+    public abstract class BaseApiService : IService, IRequiresRequest
     {
+        public BaseApiService(
+            ILogger logger,
+            IServerConfigurationManager serverConfigurationManager,
+            IHttpResultFactory httpResultFactory)
+        {
+            Logger = logger;
+            ServerConfigurationManager = serverConfigurationManager;
+            ResultFactory = httpResultFactory;
+        }
+
         /// <summary>
-        /// Gets or sets the logger.
+        /// Gets the logger.
         /// </summary>
         /// <value>The logger.</value>
-        public ILogger Logger => ApiEntryPoint.Instance.Logger;
+        protected ILogger Logger { get; }
+
+        /// <summary>
+        /// Gets or sets the server configuration manager.
+        /// </summary>
+        /// <value>The server configuration manager.</value>
+        protected IServerConfigurationManager ServerConfigurationManager { get; }
 
         /// <summary>
-        /// Gets or sets the HTTP result factory.
+        /// Gets the HTTP result factory.
         /// </summary>
         /// <value>The HTTP result factory.</value>
-        public IHttpResultFactory ResultFactory => ApiEntryPoint.Instance.ResultFactory;
+        protected IHttpResultFactory ResultFactory { get; }
```

Comment:
again, why?

Reply:
Can’t (easily) access the instance of ApiEntryPoint in testing code.